cmd/program/program.go

```diff
@@ -524,16 +617,21 @@ func (p *Project) CreateFileWithInjection(pathToCreate string, projectPath strin
 		createdTemplate := template.Must(template.New(fileName).Parse(string(p.FrameworkMap[p.ProjectType].templater.ServerWithDB())))
 		err = createdTemplate.Execute(createdFile, p)
 	case "routes":
-		createdTemplate := template.Must(template.New(fileName).Parse(string(p.FrameworkMap[p.ProjectType].templater.Routes())))
+		if p.AdvancedOptions["AddHTMXTempl"] {
+			p.CreateTemplateImports()
+			p.CreateTemplateRoutes()
+		}
+		routeFileBytes := p.FrameworkMap[p.ProjectType].templater.Routes()
+		createdTemplate := template.Must(template.New(fileName).Parse(string(routeFileBytes)))
 		err = createdTemplate.Execute(createdFile, p)
 	case "file1":
-		createdTemplate := template.Must(template.New(fileName).Parse(string(p.WorkflowMap[p.Workflow].templater.File_1())))
+		createdTemplate := template.Must(template.New(fileName).Parse(string(advanced.File_1())))
 		err = createdTemplate.Execute(createdFile, p)
 	case "file2":
-		createdTemplate := template.Must(template.New(fileName).Parse(string(p.WorkflowMap[p.Workflow].templater.File_2())))
+		createdTemplate := template.Must(template.New(fileName).Parse(string(advanced.File_2())))
 		err = createdTemplate.Execute(createdFile, p)
 	case "file3":
```

Comment:
i know this isnt your PR but i dont like the fileX casing here

Reply:
I have fixed this, initial idea was to use this logic for other file injections